src/Jobs/Notifications/ProductNotificationJob.php

```diff
@@ -121,6 +122,21 @@ protected function get_after_notification_status( string $topic ): string {
 		}
 	}
 
+	/**
+	 * If there is a valid Item ID and topic is a deletion topic. Mark the coupon as unsynced.
+	 *
+	 * @param string $topic
+	 * @param int    $item
+	 */
+	protected function maybe_mark_as_unsynced( string $topic, int $item ): void {
+		if ( ! str_contains( $topic, '.delete' ) ) {
```

Comment:
I was going to ask the same question like https://github.com/woocommerce/google-listings-and-ads/pull/2278#discussion_r1504297176 but I saw you had a reply there. Maybe we could add a TODO comment for it?